src/nvim/ex_docmd.c

```diff
@@ -1335,25 +1419,109 @@ static char_u * do_one_cmd(char_u **cmdlinep,
    * is equal to the lower.
    */
 
+  // ea.addr_type for user commands is set by find_ucmd
+  if (!IS_USER_CMDIDX(ea.cmdidx)) {
+    if (ea.cmdidx != CMD_SIZE) {
+      ea.addr_type = cmdnames[(int)ea.cmdidx].cmd_addr_type;
+    } else {
+      ea.addr_type = ADDR_LINES;
+    }
+    // :wincmd range depends on the argument
+    if (ea.cmdidx == CMD_wincmd) {
```

Comment:
I bundled the skipwhite part of that patch (don't remember why, but I probably didn't notice 580 at all) into 572. I'll add this check there and change the commit description.